department-of-veterans-affairs/caseflow
# Task serialization: return name of inactive organization

Resolves [Sentry alert](https://dsva.slack.com/archives/CLRTL92TW/p1632240728192700)

### Description
When `task.assigned_to.inactive = true`, show the assignee info instead of raising an error.

### Acceptance Criteria
- [x] Doesn't error on inactive organization assignee

### Testing Plan
See RSpec

1. Go to Case Details for an appeal
2. Pick a task assigned to an organization
3. Make the organization inactive
4. Reload Case Details. Before the PR, note the error; after the PR, no error and the assignee name is shown.

### Reflection

BLUF: stay clear away from `default_scope` if you ever want to refer to records excluded by the default scope.

* `default_scope` is used for Organization, DecisionIssue, and RequestDecisionIssue
* Appropriately used by DecisionIssue and RequestDecisionIssue for soft deleting records.
* For Organization, tasks can refer to inactive orgs, which results in a nil or erroneous `task.assigned_to`
* `acts_as_paranoid` is also used to soft-delete records. It uses `default_scope`: https://github.com/ActsAsParanoid/acts_as_paranoid/blob/cf7c83f083fc87147ba272b380c2b28469a704a8/lib/acts_as_paranoid.rb#L49
